conan/tools/cmake/cmakedeps/templates/config.py

```diff
@@ -47,25 +47,33 @@ def template(self):
 
         {% if is_module %}
         include(FindPackageHandleStandardArgs)
-        set({{ pkg_name }}_FOUND 1)
-        set({{ pkg_name }}_VERSION "{{ version }}")
+        set({{ file_name }}_FOUND 1)
+        set({{ file_name }}_VERSION "{{ version }}")
 
-        find_package_handle_standard_args({{ pkg_name }}
-                                          REQUIRED_VARS {{ pkg_name }}_VERSION
-                                          VERSION_VAR {{ pkg_name }}_VERSION)
-        mark_as_advanced({{ pkg_name }}_FOUND {{ pkg_name }}_VERSION)
+        find_package_handle_standard_args({{ file_name }}
+                                          REQUIRED_VARS {{ file_name }}_VERSION
+                                          VERSION_VAR {{ file_name }}_VERSION)
+        mark_as_advanced({{ file_name }}_FOUND {{ file_name }}_VERSION)
         {% endif %}
 
         include(${CMAKE_CURRENT_LIST_DIR}/cmakedeps_macros.cmake)
         include(${CMAKE_CURRENT_LIST_DIR}/{{ targets_include_file }})
         include(CMakeFindDependencyMacro)
 
         foreach(_DEPENDENCY {{ '${' + pkg_name + '_FIND_DEPENDENCY_NAMES' + '}' }} )
+            # Check that we have not already called a find_package with the transitive dependency
             if(NOT {{ '${_DEPENDENCY}' }}_FOUND)
+            {% if is_module %}
+                find_dependency({{ '${_DEPENDENCY}' }} REQUIRED MODULE)
+            {% else %}
                 find_dependency({{ '${_DEPENDENCY}' }} REQUIRED NO_MODULE)
+            {% endif %}
             endif()
         endforeach()
 
+        set({{ file_name }}_INCLUDE_DIRS {{ '${' + pkg_name + '_INCLUDE_DIRS' + config_suffix + '}' }} )
+        set({{ file_name }}_LIBRARIES {{ '${' + pkg_name + '_LIBRARIES' + config_suffix + '}' }} )
```

Comment:
I don't know if they are defined elsewhere, but it would be very nice to add:
`<file_name>_INCLUDE_DIR>` (no `S`)
`<file_name>_DEFINITIONS>`
`<file_name>_VERSION_STRING>`

Grep `_create_cmake_module_variables` in CCI, you'll see some boilerplate related to these variables.

Reply:
Can we easily add these @czoido ?

Reply:
Yes, I'll add those 👍